connection/s3.go

```diff
@@ -0,0 +1,60 @@
+package connection
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/flanksource/commons/collections"
+)
+
+// +kubebuilder:object:generate=true
+type S3Connection struct {
+	AWSConnection `yaml:",inline"`
+	Bucket        string `yaml:"bucket,omitempty" json:"bucket,omitempty"`
+	// glob path to restrict matches to a subset
+	ObjectPath string `yaml:"objectPath,omitempty" json:"objectPath,omitempty"`
```

Comment:
I'm not entirely sure if this should be part of the connection spec or if it's something that's specific to the s3 canary check.